packages/protocol/test-sol/integration/Integration.t.sol

```diff
@@ -1,19 +1,81 @@
 // SPDX-License-Identifier: UNLICENSED
-pragma solidity ^0.5.13;
+pragma solidity >=0.5.13 <0.8.20;
```

Comment:
Can we change it to 0.8 only ?

Reply:
Yes totally. The integration tests run and pass as expected on `0.8.0`.

For context, I matched the version @soloseng specified in https://github.com/celo-org/celo-monorepo/pull/10995:

https://github.com/celo-org/celo-monorepo/blob/d931ead1c1f508d1380ddc99d4339a2e51896020/packages/protocol/test-sol/constants.sol#L2

I decided to match the versions defined here:

https://github.com/celo-org/celo-monorepo/blob/c27a53eaf1cd5729e1b785e90051eac3fb821184/packages/protocol/test-sol/constants.sol#L1-L2

Two questions: 
1. @soloseng: Is there a specific reason you chose `<0.8.2` over `<0.8.0`?
2. @pahor167: Is there a specific you prefer `<0.8.0` over `<0.8.2`?

Reply:
Good point, hanks for clarifying. 
Unfortunately, I have to use a lower bound of `>=0.5.13`, because I'm using `IProxy.sol` which is on `>=0.5.13`

https://github.com/celo-org/celo-monorepo/blob/19faa9ba6b7a80590ea4c23173d5aeedd45f891e/packages/protocol/test-sol/integration/Integration.t.sol#L10-L11

https://github.com/celo-org/celo-monorepo/blob/19faa9ba6b7a80590ea4c23173d5aeedd45f891e/packages/protocol/contracts/common/interfaces/IProxy.sol#L1-L4

Am I missing something or does this make sense to you as well?

Reply:
You should be able to use IProxy with 0.8.* without any issues since it is still within range of >=0.5.13 <0.9.0

Reply:
Hmm, in that case I'll have to look into why the contracts don't compile with 

```sol
pragma solidity >=0.8.7 <0.8.20;
```

This is the compilation error I get:

```sh
./migrations_sol/run_integration_tests_in_anvil.sh

[⠃] Compiling...2024-06-04T15:37:08.987061Z ERROR foundry_compilers::resolver: failed to resolve versions
[⠊] Compiling...
Error:
Found incompatible Solidity versions:
test-sol/integration/Integration.t.sol (>=0.8.7 <0.8.20) imports:
    lib/celo-foundry/src/Test.sol (^0.5.13)
    test-sol/utils.sol (^0.5.13)
    test-sol/constants.sol (>=0.5.13 <0.8.20)
    contracts/common/interfaces/IRegistry.sol (>=0.5.13 <0.9.0)
    contracts/common/interfaces/IProxy.sol (>=0.5.13 <0.9.0)
    lib/celo-foundry/lib/forge-std/src/Test.sol (>=0.5.13 <0.9.0)
    lib/celo-foundry/src/Precompiles.sol (^0.5.13)
    lib/celo-foundry/src/PrecompileHandler.sol (^0.5.13)
    lib/celo-foundry/src/Test.sol (^0.5.13)
    lib/openzeppelin-contracts/contracts/utils/EnumerableSet.sol (^0.5.0)
    lib/celo-foundry/lib/forge-std/src/Script.sol (>=0.5.13 <0.9.0)
    lib/celo-foundry/lib/forge-std/lib/ds-test/src/test.sol (>=0.5.0)
    lib/celo-foundry/lib/forge-std/src/Vm.sol (>=0.5.13 <0.9.0)
    lib/celo-foundry/lib/forge-std/src/console2.sol (>=0.4.22 <0.9.0)
    lib/celo-foundry/src/Precompiles.sol (^0.5.13)
    lib/celo-foundry/lib/forge-std/src/Vm.sol (>=0.5.13 <0.9.0)
    lib/celo-foundry/lib/forge-std/src/console.sol (>=0.4.22 <0.9.0)
    lib/celo-foundry/lib/forge-std/src/console2.sol (>=0.4.22 <0.9.0)
```

Reply:
> Is there a specific reason you chose <0.8.20 over <0.8.0?

I used <0.8.20, so that it matches the other contracts we have that are using 0.8.* versions. This helps avoid the version incompatibility. 

I think I had previously encountered and error when a contract was compiled using 0.8.20.

Reply:
This might be due to the `import "celo-foundry/Test.sol";` import. try switching to `celo-foundry-8`

Reply:
Looks like the reason I'm currently bound to ` >=0.5.13 <0.8.20` is because

1. `Integration.t.sol` depends on `utils.sol`:

    https://github.com/celo-org/celo-monorepo/blob/d31c336461ea07ddb7a54b42545996b71a512ec5/packages/protocol/test-sol/integration/Integration.t.sol#L6-L6

1. `utils.sol` is used in various tests that are bound to the `0.5.x` range with `^0.5.13`
    
    https://github.com/celo-org/celo-monorepo/blob/d31c336461ea07ddb7a54b42545996b71a512ec5/packages/protocol/test-sol/utils.sol#L1-L1

    like 
    
    ```
    Found incompatible Solidity versions:
    
    test-sol/common/Proxy.t.sol (^0.5.13)
    test-sol/governance/network/BlockchainParameters.t.sol (^0.5.13)
    test-sol/governance/network/EpochRewards.t.sol (^0.5.13)
    test-sol/governance/validators/DowntimeSlasher.t.sol (^0.5.13)
    test-sol/governance/validators/Validators.t.sol (^0.5.13)
    test-sol/identity/Random.t.sol (^0.5.13)
    test-sol/integration/RevokeCeloAfterL2Transition.sol (^0.5.13)
    test-sol/identity/Random.t.sol (^0.5.13)
    test-sol/voting/Election.t.sol (^0.5.13)
    ```

I can try to modify the test ranges to `>=0.5.13 <0.8.20`, but I don't completely understand the implications yet:

```diff
- ^0.5.13
+ >=0.5.13 <0.8.20
```

Reply:
Unfortunately, this is a downside to using different pragma version concurrently. Modifying the test contracts to use `>=0.5.13 <0.8.20` would most likely cause errors, as these contracts import the core contracts that are set to `0.5.13` 😔.

Short of creating a version of the Utils contract in 0.8, i'm not sure what other fix would work.